tests/system/model_monitoring/test_model_monitoring.py

```diff
@@ -403,11 +423,134 @@ def test_model_monitoring_voting_ensemble(self):
+        )
+
+        # Define feature vector and save it to MLRun's feature store DB
+        fv = mlrun.feature_store.FeatureVector(
+            "diabetes-features",
+            features=["diabetes-set.*"],
+            label_feature="target-set.target",
+        )
+        fv.save()
+
+        assert (
+            fv.uri == f"store://feature-vectors/{self.project_name}/diabetes-features"
+        )
+
+        # Request (get or create) the offline dataset from the feature store and save to a parquet target
+        mlrun.feature_store.get_offline_features(
+            fv, target=mlrun.datastore.targets.ParquetTarget()
+        )
+
+        # Train the model using the auto trainer from the marketplace
+        train = mlrun.import_function("hub://auto_trainer", new_name="train")
+        train.deploy()
+        model_class = "sklearn.linear_model.LinearRegression"
+        model_name = "diabetes_model"
+        label_columns = "target"
+
+        train_run = train.run(
+            inputs={"dataset": fv.uri},
+            params={
+                "model_class": model_class,
+                "model_name": model_name,
+                "label_columns": label_columns,
+                "train_test_split_size": 0.2,
+            },
+            handler="train",
+        )
+
+        # Remove features from model obj and set feature vector uri
+        db = mlrun.get_run_db()
+        model_obj: mlrun.artifacts.ModelArtifact = (
+            mlrun.datastore.store_resources.get_store_resource(
+                train_run.outputs["model"], db=db
+            )
+        )
+        model_obj.inputs = []
+        model_obj.feature_vector = fv.uri + ":latest"
+        mlrun.artifacts.model.update_model(model_obj)
+
+        # Set the serving topology to simple model routing
+        # with data enrichment and imputing from the feature vector
+        serving_fn = mlrun.import_function("hub://v2_model_server", new_name="serving")
+        serving_fn.set_topology(
+            "router",
+            mlrun.serving.routers.EnrichmentModelRouter(
+                feature_vector_uri=str(fv.uri), impute_policy={"*": "$mean"}
+            ),
+        )
+        serving_fn.add_model("diabetes_model", model_path=train_run.outputs["model"])
+
+        # Enable model monitoring
+        serving_fn.set_tracking()
+
+        # Deploy the serving function
+        serving_fn.deploy()
+
+        # Invoke the model
+        fset = mlrun.feature_store.get_feature_set("diabetes-set")
+        res = fset.to_dataframe()
+        data_to_invoke = res.reset_index()["patient_id"].values
+        for msg_batch in range(51):
+            ids_for_prediction = [[str(choice(data_to_invoke))] for i in range(5)]
+            serving_fn.invoke(
+                path="/v2/models/diabetes_model/infer",
+                body={"inputs": ids_for_prediction},
+            )
+            sleep(0.2)
+
+        # Sleep to allow TSDB to be written (10/m)
+        sleep(20)
+
```

Comment:
we mostly try to avoid using sleep, so use it only if really required, our system tests are already pretty long

Reply:
This part was removed (see my previous comment)